inventory_adjustment_reason/models/stock_quant.py

```diff
@@ -0,0 +1,6 @@
+from odoo import fields, models
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+    note = fields.Text(string="Inventory Adjustment Reason")
```

Comment:
This only stores the note on the stock.quant.

If I make a second adjustment the original note is lost.

Assuming they want to keep the note, this needs to be persisted to the stock.move when the quant is applied.

The new field would also need to be displayed on the stock.move form.

(I've not tested this code).

```suggestion
class StockQuant(models.Model):
    _inherit = "stock.quant"
    note = fields.Text(string="Inventory Adjustment Reason")
    
    def _get_inventory_move_values(self, qty, location_id, location_dest_id, out=False):
        res = super()._get_inventory_move_values(
            qty, location_id, location_dest_id, out=out
        )

        res.update(
            {
                "note": self.note,
            }
        )

        return res

    def _apply_inventory(self):
        res = super()._apply_inventory()

        self.write({"note": False})

        return res

    @api.model
    def _get_inventory_fields_write(self):
        res = super()._get_inventory_fields_write()
        res.append("note")
        return res


class StockMove(models.Model):
    _inherit = "stock.move"

    note = fields.Text()
```